ValidationLibrary.Rules.Tests/BaseRuleTests.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Threading.Tasks;
+using NSubstitute;
+using NUnit.Framework;
+using Octokit;
+
+namespace ValidationLibrary.Tests.Rules
+{
+    [TestFixture]
+    public abstract class BaseRuleTests<T> where T : IValidationRule
+    {
+        protected User Owner { get; set; }
+        protected T _rule;
+
+        protected IGitHubClient MockClient { get; set; }
+        protected IRepositoriesClient MockRepositoryClient { get; set; }
+        protected IRepositoryContentsClient MockRepositoryContentClient { get; set; }
+
+        protected const string MainBranch = "master";
+
+        [SetUp]
+        public async Task Setup()
+        {
+            Owner = new User(null, null, null, 0, null, DateTime.UtcNow, DateTime.UtcNow, 0, null, 0, 0, false, null, 0, 0, null, "protacon", "protacon", null, 0, null, 0, 0, 0, null, new RepositoryPermissions(), false, null, null);
+
+            MockClient = Substitute.For<IGitHubClient>();
+            MockRepositoryClient = Substitute.For<IRepositoriesClient>();
+            MockClient.Repository.Returns(MockRepositoryClient);
+            MockRepositoryContentClient = Substitute.For<IRepositoryContentsClient>();
+            MockRepositoryClient.Content.Returns(MockRepositoryContentClient);
+            OnSetup();
+            await _rule.Init(MockClient);
+        }
+
+        protected abstract void OnSetup();
+
+        [Test]
+        public async Task Init_DoesntThrow()
+        {
+            await _rule.Init(MockClient);
```

Comment:
I think here could be assert does not throw async

Reply:
This whole test is no longer necessary.